packages/@aws-cdk/aws-apigateway/lib/gateway-response.ts

```diff
@@ -0,0 +1,199 @@
+import { Construct, IResource, Resource } from '@aws-cdk/core';
+import { CfnGatewayResponse } from './apigateway.generated';
+import { IRestApi } from './restapi';
+
+/**
+ * Represents gateway response resource.
+ */
+export interface IGatewayResponse extends IResource {
+}
+
+/**
+ * Properties for a new gateway response.
+ */
+export interface GatewayResponseProps extends GatewayResponseOptions {
+  /**
+   * Rest api resource to target.
+   */
+  readonly restApi: IRestApi;
+}
+
+/**
+ * Options to add gateway response.
+ */
+export interface GatewayResponseOptions {
+  /**
+   * Response type to associate with gateway response.
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html
+   */
+  readonly type: ResponseType;
+
+  /**
+   * Http status code for response.
+   * @default - standart http status code for the response type.
+   */
+  readonly statusCode?: string;
+
+  /**
+   * Custom path, query and headers for response.
+   * @default - No custom paramter will be added to response.
+   */
+  readonly parameters?: { [key: string]: string };
```

Comment:
Need to be addressed per https://github.com/aws/aws-cdk/pull/7441#discussion_r414393791